server/start.go

```diff
@@ -258,10 +258,10 @@ func startInProcess(ctx *Context, clientCtx client.Context, appCreator types.App
 	// service if API or gRPC is enabled, and avoid doing so in the general
 	// case, because it spawns a new local tendermint RPC client.
 	if config.API.Enable || config.GRPC.Enable {
-		clientCtx = clientCtx.
-			WithClient(local.New(tmNode))
+		clientCtx = clientCtx.WithClient(local.New(tmNode))
 
 		app.RegisterTxService(clientCtx)
+		app.RegisterTendermintService(clientCtx)
```

Comment:
Why don't we just create this as a module? I understand it doesn't implement all the methods a module does not store data but it does register a service. In my mind, anything that registers services should be a module. This to me begs the question of how the module interface is designed. Anyway, this is a somewhat separate discussion from this PR... no need to change anything now, just pointing out.

Reply:
Interesting. For me, a module is a mini state-machine, so it must have state.

I would see this service (or the TxService just above) as "optional service plugins".